### ISSUE TITLE
Git-flow?

### ISSUE THREAD

**Issue by AzureKitsune** [MahApps/MahApps.Metro#574]
Lately, I've noticed that people and applications were picking in the [git-flow](http://nvie.com/posts/a-successful-git-branching-model/) model.

The TL;DR would be: 4-5 branches:
- master branch: When code is commit here, it's a new release
- develop branch: Code here is code for the /next/ release.
  - features branches: Branches that contain new features which will eventually be merged into develop.
  - hotfix branches: Same as feature branches, except they are for bug fixes.
  - release branches: Branches for updating the version number and maybe applying last minute bug fixes. This gets merged into both master and develop so that the current release and the next release receive the bug fix.

Obviously, the link I provided explains things a little better.

@shiftkey Perhaps such a switch can improve our workflow, but yeah, just wanted to know your opinion on this.

cc @flagbug @punker76
